miasm2/expression/expression.py

```diff
@@ -318,12 +318,10 @@ def replace_expr(self, dct=None):
         @dct: dictionary of Expr -> *
```

Comment:
After reading your patch, I wonder if the 'optional' argument is really interesting here.
I see no usecase where we want to call this function without an expression to replace.

So maybe we can change the api to have the `dct` mandatory.
What about this?

Reply:
Thats a good idea, I think.  
I also thought about using lighter version of this with only one mapping dst-src (I use it myself in my inherited expression class). This will boost perfomance a bit more for such simple replacings.